From b5096098c62adb19dbf4a39b480909766c9026e7 Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@kernel.dk>
Date: Wed, 29 Apr 2020 10:13:20 -0600
Subject: [PATCH] Use __off64_t for offsets
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

loff_t isn't available everywhere, and off_t is only 32-bits on some
archs.

Reported-by: Milan P. StaniÄ‡ <mps@arvanta.net>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
---
 src/include/liburing.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/include/liburing.h b/src/include/liburing.h
index ae5542c..2c8beef 100644
--- a/src/include/liburing.h
+++ b/src/include/liburing.h
@@ -193,8 +193,8 @@ static inline void io_uring_prep_rw(int op, struct io_uring_sqe *sqe, int fd,
 }
 
 static inline void io_uring_prep_splice(struct io_uring_sqe *sqe,
-					int fd_in, loff_t off_in,
-					int fd_out, loff_t off_out,
+					int fd_in, __off64_t off_in,
+					int fd_out, __off64_t off_out,
 					unsigned int nbytes,
 					unsigned int splice_flags)
 {
-- 
1.8.3.1

